### PR TITLE
Add support for validator auto-mapping

### PR DESCRIPTION
**Pull request by dunglas**
Adds support for symfony/symfony#27735 in the bundle.

Related config:

```yaml
framework:
    validation:
        auto_mapping:
            App\Entity\: []
```

### CODE REVIEW

File: DependencyInjection/DoctrineExtension.php [hunk 1]
```diff
@@ -760,6 +762,23 @@ private function loadPropertyInfoExtractor($entityManagerName, ContainerBuilder
         $propertyExtractorDefinition->addTag('property_info.type_extractor', ['priority' => -999]);
     }
 
+    /**
+     * Loads a validator loader for each defined entity manager.
+     *
+     * @param string $entityManagerName
+     */
+    private function loadValidatorLoader($entityManagerName, ContainerBuilder $container)
```
Comment (greg0ire) [inline DependencyInjection/DoctrineExtension.php]: This package requires php 7.1, I think you can use type hints here:

```suggestion
    private function loadValidatorLoader(string: $entityManagerName, ContainerBuilder $container): void
```

**Comment by dunglas** [comment]
I tested this PR on a real project, and it's working properly.

**Comment by Majkl578** [comment]
> and it's working properly

This will need some sort of test otherwise it can break in future. :|

**Comment by dunglas** [comment]
Yes I totally agree. At least a functional test is mandatory.

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Review by SenseException — CHANGES_REQUESTED** [review]
A phpcs issue is still there. Otherwise this is :+1:

File: DependencyInjection/DoctrineExtension.php [hunk 2]
```diff
@@ -761,6 +763,21 @@ private function loadPropertyInfoExtractor($entityManagerName, ContainerBuilder
         $propertyExtractorDefinition->addTag('property_info.type_extractor', ['priority' => -999]);
     }
 
+    /**
+     * Loads a validator loader for each defined entity manager.
+     */
+    private function loadValidatorLoader(string $entityManagerName, ContainerBuilder $container) : void
+    {
+        if (!class_exists(DoctrineLoader::class)) {
```
Comment (greg0ire) [inline DependencyInjection/DoctrineExtension.php]: This should fix the build:
```suggestion
        if (! class_exists(DoctrineLoader::class)) {
```

File: DependencyInjection/DoctrineExtension.php [hunk 3]
```diff
@@ -761,6 +763,21 @@ private function loadPropertyInfoExtractor($entityManagerName, ContainerBuilder
         $propertyExtractorDefinition->addTag('property_info.type_extractor', ['priority' => -999]);
     }
 
+    /**
+     * Loads a validator loader for each defined entity manager.
+     */
+    private function loadValidatorLoader(string $entityManagerName, ContainerBuilder $container) : void
+    {
+        if (! class_exists(DoctrineLoader::class)) {
```
Comment (stof) [inline DependencyInjection/DoctrineExtension.php]: This requires testing for LoaderInterface first as the bridge does not prevent installing its latest version without the Validator component (which is expected) and then that would be a fatal error due to the missing interface.

**Review by stof — CHANGES_REQUESTED** [review]

**Review by SenseException — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Comment by dunglas** [comment]
Thanks a lot for finishing this one Andreas!

**Comment by tarjei** [comment]
It seems that the blogpost for this feature has not been updated with the info about auto enabling it:
https://symfony.com/blog/new-in-symfony-4-3-automatic-validation

**Pull request merged**